src/game/sprite.ts

```diff
@@ -0,0 +1,63 @@
+import Resources from './resources';
+
+export default class Sprite {
+  private _index: number;
+  url: string;
+  pos: [number, number];
+  size: [number, number];
+  speed?: number;
+  frames?: [number, number];
+  dir?: 'horizontal' | 'vertical';
+  once?: boolean;
+  done?: boolean;
+
+  constructor(url: string, pos: [number, number], size: [number, number], speed?: number, frames?: [number, number], dir?: 'horizontal' | 'vertical', once?: boolean) {
+    this._index = 0;
+    this.url = url;
+    this.pos = pos;
+    this.size = size;
+    this.speed = speed === undefined ? 0 : speed;
+    this.frames = frames;
+    this.dir = dir;
+    this.once = once;
+  }
+
+  update(dt: number): void {
+    if (this.speed !== undefined) {
+      this._index += this.speed * dt;
+    }
+  }
+
+  render(ctx: CanvasRenderingContext2D, resources: Resources): void {
+    let  frame = 0;
```

Comment:
❗️
Посмотри по коду где ещё есть лишний пробелы, мб я пропустил где-то
```suggestion
    let frame = 0;
```